### DISCUSSION TITLE
Embedded generates Maximum stack exceeded

### DISCUSSION THREAD

**Discussion by imcyee** [mikro-orm/mikro-orm#3194] [General]
Mikroorm version: latest.
Embedded generates Maximum stack exceeded when the field name and embedded field key name are the same when Embedded having `prefix:false`

It is looping in this line,
EntityComparator.js
176:  `const getRootProperty = (prop) => prop.embedded ? getRootProperty(meta.properties[prop.embedded[0]]) : prop;`

From `meta.properties` I can see the email properties is overwritten by this. Hence, it just kept looping at line 176.
```
email: {
    name: 'email',
    reference: 'scalar',
    getter: false,
    setter: false,
    type: 'string',
    embedded: [ 'email', 'email' ],
    fieldNames: [ 'email' ],
    unsigned: undefined,
    columnTypes: [ 'varchar(255)' ]
  },
```

**Comment by imcyee** [comment]
```
@Entity()
export class User { 
  @PrimaryKey()
  userId: string
   
  @Embedded(() => EmailObject, { prefix: false })
  email: EmailObject
}

@Embeddable()
export class EmailObject {

  isEqual(subjectEmail: EmailObject) {
    return subjectEmail && (this.email === subjectEmail.email)
  }

  // weird bug of naming it exactly the same name as field will generate MAXIMUM call stack
  @Property()
  email!: string;
 
  private constructor() { }
  public static create(email: string)  { 
    const entity = new EmailObject()
    entity.email = email
  }
}

```

**Reply by B4nan** [reply]
FYI custom mapped type would serve a better purpose, this is not the right fit for embeddables.

**Reply by imcyee** [reply]
Do you mean this https://mikro-orm.io/docs/custom-types?

**Reply by imcyee** [reply]
Sorry, may I know why isn't it a right fit? 
Was using it based on the doc: `Value objects such as date range or address are the primary use case for this feature.(Embeddable)`

**Reply by B4nan** [reply]
Because you are trying to wrap a single property, there is no need for embeddables here.

**Reply by imcyee** [reply]
Oh, I see, i used the wrong example. Below one will generate Maximum stack exceeded as well. 
```
@Entity()
export class User { 
  @PrimaryKey()
  userId: string
   
  @Embedded(() => Phone, { prefix: false })
  phone: Phone
}


@Embeddable()
export class Phone {

  @Property()
  contactCountry!: string;

  @Property()
  phone!: string;

  private constructor() { }

  public static create(props) {
    // guard 

    const entity = new this()
    entity.contactCountry = props.contactCountry
    entity.phone = props.phoneNumber 
    return entity
  }
}
```

**Reply by B4nan** [reply]
Yeah I am not trying to say this is not a problem, it should indeed work :]

**Reply by imcyee** [reply]
Yaya, agree. Actually it can be solved by simply renaming the field or the property name not to be the same.

**Reply by imcyee** [reply]
Should I link this to an issue so future people can see?
